exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -213,6 +213,43 @@ func TestExporterLogs(t *testing.T) {
 		rec.WaitItems(1)
 	})
 
+	t.Run("publish with dynamic index data_stream mode", func(t *testing.T) {
```

Comment:
I think there are some parts missing from the tests
- scope attributes
- fields in doc
  - default is used when no data_stream attributes apply
  - data_stream attributes are present
- doc action
  - default is used when no data_stream attributes apply